cmd/vicadmin/server.go

```diff
@@ -75,6 +77,80 @@ const (
 	genericErrorMessage    = "Internal Server Error; see /var/log/vic/vicadmin.log for details" // for http errors that shouldn't be displayed in the browser to the user
 )
 
+//wrapper struct around net.Conn for our custom funcs
+type Conn struct {
+	net.Conn
+	b             byte
+	err           error
+	oneTimeSwitch bool
+}
+
+//Returns number of bytes read
+func (c *Conn) Read(b []byte) (int, error) {
+	//oneTimeSwitch bool of first conn starts off as true so we do the TLS check once
+	if c.oneTimeSwitch {
+		c.oneTimeSwitch = false
+		b[0] = c.b
+		// if there's more bytes to read
+		if len(b) > 1 && c.err == nil {
+			//recurse on next byte
+			n, e := c.Conn.Read(b[1:])
+			//close connection if error during reading
+			if e != nil {
+				c.Conn.Close()
+			}
+			//return total num of bytes read (+ current) and pass error e
+			return n + 1, e
+		}
+		//only one byte read
+		return 1, c.err
+	}
+	//using the default Conn read
+	return c.Conn.Read(b)
+}
+
+//wrapper struct for our custom funcs
```

Comment:
Let's make this comment a bit more descriptive. What does `ReqListener` actually allow us to do?